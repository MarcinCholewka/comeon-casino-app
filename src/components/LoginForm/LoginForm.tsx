import { Form, useFormikContext } from 'formik';
import { Button, Icon, Input } from 'semantic-ui-react';
import { isEmpty } from 'lodash';

import { TFormValues } from '@views/Login';

export const LoginForm = () => {
  const { isSubmitting, errors, handleChange, touched } =
    useFormikContext<TFormValues>();

  return (
    <Form>
      <div className='fields'>
        <div className='required field m-[10px]'>
          <Input
            error={!!(touched.username && errors.username)}
            icon={{ name: 'user' }}
            type='text'
            name='username'
            placeholder='Username'
            onChange={handleChange}
          />
          {touched.username && errors.username && (
            <div className='text-red-500 text-sm text-left pl-1'>
              {errors.username}
            </div>
          )}
        </div>
        <div className='required field m-[10px]'>
          <Input
            error={!!(touched.password && errors.password)}
            icon={{ name: 'lock' }}
            type='password'
            name='password'
            placeholder='Password'
            onChange={handleChange}
          />
          {touched.password && errors.password && (
            <div className='text-red-500 text-sm text-left pl-1'>
              {errors.password}
            </div>
          )}
        </div>
        <div className='field m-[10px]'>
          <Button
            disabled={!isEmpty(errors)}
            loading={isSubmitting}
            type='submit'>
            Login
            <Icon name='chevron right' />
          </Button>
        </div>
      </div>
    </Form>
  );
};
