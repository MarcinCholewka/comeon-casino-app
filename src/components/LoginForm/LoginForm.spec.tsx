import { render } from '@testing-library/react';
import { LoginForm } from './LoginForm';
import { Formik } from 'formik';

describe('<LoginForm />', () => {
  it('should render without crashing', () => {
    const { container, getByPlaceholderText, getByText } = render(
      <Formik
        initialValues={{ username: '', password: '' }}
        onSubmit={jest.fn()}>
        <LoginForm />
      </Formik>,
    );

    const usernameInput = getByPlaceholderText('Username');
    const passwordInput = getByPlaceholderText('Password');
    const loginButton = getByText('Login');

    expect(usernameInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
    expect(loginButton).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
