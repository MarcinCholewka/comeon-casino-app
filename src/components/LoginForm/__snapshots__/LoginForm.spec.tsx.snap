// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render without crashing 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="fields"
    >
      <div
        class="required field m-[10px]"
      >
        <div
          class="ui icon input"
        >
          <input
            name="username"
            placeholder="Username"
            type="text"
          />
          <i
            aria-hidden="true"
            class="user icon"
          />
        </div>
      </div>
      <div
        class="required field m-[10px]"
      >
        <div
          class="ui icon input"
        >
          <input
            name="password"
            placeholder="Password"
            type="password"
          />
          <i
            aria-hidden="true"
            class="lock icon"
          />
        </div>
      </div>
      <div
        class="field m-[10px]"
      >
        <button
          class="ui button"
          type="submit"
        >
          Login
          <i
            aria-hidden="true"
            class="chevron right icon"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
